enterprise/api/v1/views.py

```diff
@@ -282,6 +283,94 @@ def _validate_license_revoke_data(self, request_data):
 
         return None
 
+    @staticmethod
+    def _has_user_completed_course_run(enterprise_enrollment, course_overview):
+        """
+        Returns True if the user who is enrolled in the given course has already
+        completed this course, false otherwise.  The course may be "completed"
+        if the user earned a certificate, or if the course run has ended.
+        """
+        certificate_info = get_certificate_for_user(
+            enterprise_enrollment.enterprise_customer_user.username,
+            course_overview.get('id'),
+        ) or {}
+        course_run_status = get_course_run_status(
+            course_overview,
+            certificate_info,
+            enterprise_enrollment,
+        )
+
+        return course_run_status == CourseRunProgressStatuses.COMPLETED
+
+    def _enrollments_by_course_for_licensed_user(self, enterprise_customer_user):
+        """
+        Helper method to return a dictionary mapping course ids to EnterpriseCourseEnrollments
+        for each licensed enrollment associated with the given enterprise user.
+        """
+        licensed_enrollments = self.queryset.filter(
+            enterprise_course_enrollment__enterprise_customer_user=enterprise_customer_user
+        )
+        return {
+            enrollment.enterprise_course_enrollment.course_id: enrollment.enterprise_course_enrollment
+            for enrollment in licensed_enrollments
+        }
+
+    @staticmethod
+    def _mark_licensed_enrollment_as_revoked(enterprise_enrollment):
+        """
+        Marks the licensed enrollment associated with the given enterprise enrollment as revoked
+        and saves the enterprise enrollment for later.
+        """
+        licensed_enrollment = enterprise_enrollment.license
+        licensed_enrollment.is_revoked = True
+        licensed_enrollment.save()
+
+        enterprise_enrollment.saved_for_later = True
+        enterprise_enrollment.save()
+
+    def _revoke_enrollment(self, enrollment_api_client, enterprise_enrollment, course_overview):
```

Comment:
nit: it might be just because I'm new to this area of the code, but I feel like having one of these methods detailing the type of the params (mainly `course_overview`) in the docstring could be helpful